### PR TITLE
Use early return pattern to avoid nested conditions

### LINKED ISSUE #919: Use early-return pattern avoids nested conditions
## Summary

> An early-return pattern is when a function or method quickly returns a result based on a condition and finishes executing its logic.

I found nested conditions in [/api/v1/ApiSelfCheckController/](https://github.com/fosslight/fosslight/blob/212cca8a80de4957378193dd88b18b98430487eb/src/main/java/oss/fosslight/api/controller/v1/ApiSelfCheckController.java#L135-L225)
Nested conditions can cause poor readability, maintenance difficulties, difficulty understanding logic flow, and testing difficulties.
I think early-return can solve these problems.

**AS-IS**
```java
  if (searchFlag) {
    if (ossReport != null) {
        // ...
    } else {
        return responseService.getFailResult(CoConstDef.CD_OPEN_API_PARAMETER_ERROR_MESSAGE
                , CoCodeManager.getCodeString(CoConstDef.CD_OPEN_API_MESSAGE, CoConstDef.CD_OPEN_API_PARAMETER_ERROR_MESSAGE));
    }
} else {
    return responseService.getFailResult(CoConstDef.CD_OPEN_API_PERMISSION_ERROR_MESSAGE
            , CoCodeManager.getCodeString(CoConstDef.CD_OPEN_API_MESSAGE, CoConstDef.CD_OPEN_API_PERMISSION_ERROR_MESSAGE));
}
```

**TO-BE**
```java
 if (!searchFlag) {
    return responseService.getFailResult(CoConstDef.CD_OPEN_API_PERMISSION_ERROR_MESSAGE
            , CoCodeManager.getCodeString(CoConstDef.CD_OPEN_API_MESSAGE, CoConstDef.CD_OPEN_API_PERMISSION_ERROR_MESSAGE));
}

if (ossReport == null) {
    return responseService.getFailResult(CoConstDef.CD_OPEN_API_PARAMETER_ERROR_MESSAGE
            , CoCodeManager.getCodeString(CoConstDef.CD_OPEN_API_MESSAGE, CoConstDef.CD_OPEN_API_PARAMETER_ERROR_MESSAGE));
}
    // ...

```

## References
https://medium.com/swlh/return-early-pattern-3d18a41bba8

### PR DESCRIPTION
**Pull request by parkcoldroad**
## Description
Closes #919 
> An early-return pattern is when a function or method quickly returns a result based on a condition and finishes executing its logic.

I found nested conditions in [/api/v1/ApiSelfCheckController/](https://github.com/fosslight/fosslight/blob/212cca8a80de4957378193dd88b18b98430487eb/src/main/java/oss/fosslight/api/controller/v1/ApiSelfCheckController.java#L135-L225)
Nested conditions can cause poor readability, maintenance difficulties, difficulty understanding logic flow, and testing difficulties.
I think early-return can solve these problems.

**AS-IS**
```java
if (searchFlag) {
  if (ossReport != null) {
    // …
  } else {
    return responseService.getFailResult(CoConstDef.CD_OPEN_API_PARAMETER_ERROR_MESSAGE
       , CoCodeManager.getCodeString(CoConstDef.CD_OPEN_API_MESSAGE, CoConstDef.CD_OPEN_API_PARAMETER_ERROR_MESSAGE));
  }
} else {
  return responseService.getFailResult(CoConstDef.CD_OPEN_API_PERMISSION_ERROR_MESSAGE
     , CoCodeManager.getCodeString(CoConstDef.CD_OPEN_API_MESSAGE, CoConstDef.CD_OPEN_API_PERMISSION_ERROR_MESSAGE));
 }
```

**TO-BE**
```java
if (!searchFlag) {
  return responseService.getFailResult(CoConstDef.CD_OPEN_API_PERMISSION_ERROR_MESSAGE,
    CoCodeManager.getCodeString(CoConstDef.CD_OPEN_API_MESSAGE, CoConstDef.CD_OPEN_API_PERMISSION_ERROR_MESSAGE));
}

if (ossReport == null) {
  return responseService.getFailResult(CoConstDef.CD_OPEN_API_PARAMETER_ERROR_MESSAGE,
    CoCodeManager.getCodeString(CoConstDef.CD_OPEN_API_MESSAGE, CoConstDef.CD_OPEN_API_PARAMETER_ERROR_MESSAGE));
}
// ...
```

## References
https://medium.com/swlh/return-early-pattern-3d18a41bba8

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [x] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/oss/fosslight/api/controller/v1/ApiSelfCheckController.java [hunk 1]
```diff
@@ -62,177 +62,191 @@
...
+			if (ossReport == null) {
+				return responseService.getFailResult(CoConstDef.CD_OPEN_API_PARAMETER_ERROR_MESSAGE
+					, CoCodeManager.getCodeString(CoConstDef.CD_OPEN_API_MESSAGE,
+						CoConstDef.CD_OPEN_API_PARAMETER_ERROR_MESSAGE));
+			}
+
+			UploadFile bean = apiFileService.uploadFile(ossReport); // file 등록 처리 이후 upload된 file정보를 return함.
+			List<UploadFile> list = new ArrayList<>();
+			list.add(bean);
+			ArrayList<Object> checkFileLimit;
```
Comment (70825) [inline src/main/java/oss/fosslight/api/controller/v1/ApiSelfCheckController.java:169]:
```suggestion
			List<Object> checkFileLimit;
```

File: src/main/java/oss/fosslight/api/controller/v1/ApiSelfCheckController.java [hunk 2]
```diff
@@ -62,177 +62,191 @@
...
+				errorMsg = (String)result.get("errorMessage");
+			}
+
+			List<ProjectIdentification> ossComponents = (List<ProjectIdentification>)result.get("ossComponents");
+			List<List<ProjectIdentification>> ossComponentsLicense = (List<List<ProjectIdentification>>)result.get(
+				"ossComponentLicense");
+
+			if (!isEmpty(errorMsg)) {
+				throw new Exception(); // readData시 문제가 발생할 경우 parameter error로 return 함.
+			}
```
Comment (70825) [inline src/main/java/oss/fosslight/api/controller/v1/ApiSelfCheckController.java:207]:
```suggestion
			if (!isEmpty(errorMsg)) {
				throw new Exception(); // readData시 문제가 발생할 경우 parameter error로 return 함.
			}

			List<ProjectIdentification> ossComponents = (List<ProjectIdentification>)result.get("ossComponents");
			List<List<ProjectIdentification>> ossComponentsLicense = (List<List<ProjectIdentification>>)result.get(
				"ossComponentLicense");
```

**Review by 70825 — COMMENTED** [review]
This is good code 👍
However, could you please modify the following for cleaner code?

**Comment by parkcoldroad** [comment]
> This is good code 👍 However, could you please modify the following for cleaner code?

Thanks for looking. I'll fix it right away!

**Review by 70825 — APPROVED** [review]
LGTM 👍

**Review by jongwooo — APPROVED** [review]
@parkcoldroad LGTM. Thank you for your contribution!

**Review by 70825 — COMMENTED** [review]
Can you make commit messages 50 characters or less in length?

**Comment by parkcoldroad** [comment]
> Can you make commit messages 50 characters or less in length?

Sure. 😆

**Comment by soimkim** [comment]
@parkcoldroad 님, 리뷰가 늦어 죄송합니다. 
늦은 리뷰로 인해...충돌이 생겨버렸습니다. 충돌 해결해주시면 정말 정말 감사합니다. 🙇

**Comment by soimkim** [comment]
추후 줄 바꿈 수정은 별도 커밋으로 해주시면 리뷰에 좀 더 좋겠습니다. 
기여해주셔서 감사합니다.

**Pull request merged**